backend/src/services/integrationService.ts

```diff
@@ -133,27 +135,88 @@ export default class IntegrationService {
     return IntegrationRepository.findAllByPlatform(platform, this.options)
   }
 
+  static isCodePlatform(value: string): value is CodePlatform {
+    return [
+      PlatformType.GITHUB,
+      PlatformType.GITHUB_NANGO,
+      PlatformType.GITLAB,
+      PlatformType.GIT,
+      PlatformType.GERRIT,
+    ].includes(value as PlatformType)
+  }
+
   async create(data, transaction?: any, options?: IRepositoryOptions) {
     try {
       const record = await IntegrationRepository.create(data, {
         ...(options || this.options),
         transaction,
       })
 
+      const collectionService = new CollectionService(this.options)
+
+      const [insightsProject] = await collectionService.findInsightsProjectsBySegmentId(
+        record.segmentId,
+      )
+
+      if (insightsProject) {
+        this.updateInsightsProject(insightsProject.id, record.segmentId)
+      }
+
       return record
     } catch (error) {
       SequelizeRepository.handleUniqueFieldError(error, this.options.language, 'integration')
       throw error
     }
   }
 
+  private async updateInsightsProject(insightsProjectId: string, segmentId: string) {
+    const collectionService = new CollectionService(this.options)
+
+    const data: Partial<ICreateInsightsProject> = {}
+    const { platforms, widgets } = await collectionService.findSegmentsWidgetsById(segmentId)
+    data.widgets = widgets
+
+    if (platforms.some(IntegrationService.isCodePlatform)) {
+      const repositories = await collectionService.findRepositoriesForSegment(segmentId)
+      data.repositories = [
+        ...new Set(Object.values(repositories).flatMap((entries) => entries.map((e) => e.url))),
+      ]
```

Comment:
We should deduplicate repositories here, because repositories from git and github might be the same. As we usually track all repositories from both platforms

Reply:
Oops, you're absolutely right.